Change conda related sections in the README and the docs

adam-urbanczyk opened the pull request:
Small improvements of the conda installation instructions.

Jojain commented on doc/quickstart.rst:
```diff
@@ -8,10 +8,10 @@ CadQuery QuickStart
 Want a quick glimpse of what CadQuery can do?  This quickstart will demonstrate the basics of CadQuery using a simple example
 
-Prerequisites: Anaconda/Miniconda + CQ-editor running from an environment
+Prerequisites: conda + CQ-editor running from an environment
```
Sharing what goes through my head, but couldn't it be possible to setup a [binder](https://mybinder.org/) environment to give quick example of CQ to people ?

adam-urbanczyk replied:
I think that we tried, but it was really slow

Jojain replied:
Ah ok didn't know, nevermind then

jmwright commented on doc/installation.rst:
```diff
@@ -5,11 +5,29 @@ Installing CadQuery
 CadQuery 2 is based on
 `OCP <https://github.com/CadQuery/OCP>`_,
-which is a set of Python bindings for the open-source `OpenCascade <http://www.opencascade.com/>`_ modelling kernel.
+which is a set of Python bindings for the open-source ` OpenCascade <http://www.opencascade.com/>`_ modelling kernel.
```
Why the space in front of "OpenCascade" here?

adam-urbanczyk replied:
Typo, fixed

jmwright commented:
@adam-urbanczyk This PR looks good to me except that I haven't used miniforge, so I don't know it's strengths/weaknesses vs miniconda.

adam-urbanczyk commented:
@jmwright miniforge is a minimal conda installation configured to use the (free) conda-forge channel (which is the channel we are relying on). I also added this based on the comments on bloat floating around HaD/HN. I also added section on portable win install.

jmwright commented:
Ok, thanks @adam-urbanczyk  .

@Jojain since you're already involved in this discussion, will you be the other official reviewer? We normally try to have two devs check a PR before merging unless it's a super simple one.

Jojain commented:
Sure if you want too, I will try this new installation process

jmwright commented:
@Jojain If you have time, please do.

Jojain commented:
@adam-urbanczyk 
I just tried the miniforge commands lines without success.
`curl.exe -o miniforge.exe https://superuser.com` seems to install something but without more info (I don't know what is curl)

Then
`start /wait "" miniforge.exe /InstallationType=JustMe /RegisterPython=0 /NoRegistry=1 /NoScripts=1 /S /D=%USERPROFILE%\Miniforge3
` Fails with : System cannot execute specified program

jmwright commented:
It installs fine for me in Ubuntu, so maybe the problem is specific to Windows?

jmwright commented on README.md:
```diff
@@ -58,17 +58,27 @@ conda install -c conda-forge -c cadquery cadquery=master
+bash miniforge.sh -b -p $HOME/miniforge
 
 # To activate and use Miniconda
 source $HOME/miniconda/bin/activate
```
I think that `miniconda` in this line has to change to `miniforge`.

jmwright commented on doc/installation.rst:
```diff
@@ -7,9 +7,27 @@ CadQuery 2 is based on
+        bash miniforge.sh -b -p $HOME/miniforge
+
+        # To activate and use Miniconda
+        source $HOME/miniconda/bin/activate
```
Same comment as above. I think that `miniconda` in this line has to change to `miniforge`.

adam-urbanczyk replied:
done

jmwright commented:
The installation of miniforge works on Ubuntu, but the environment solve always fails when I try to run the following line.
```
conda install -c conda-forge -c cadquery cadquery=master
```

adam-urbanczyk commented:
> @adam-urbanczyk I just tried the miniforge commands lines without success. `curl.exe -o miniforge.exe https://superuser.com` seems to install something but without more info (I don't know what is curl)
> 
> Then `start /wait "" miniforge.exe /InstallationType=JustMe /RegisterPython=0 /NoRegistry=1 /NoScripts=1 /S /D=%USERPROFILE%\Miniforge3 ` Fails with : System cannot execute specified program

I made a stupid copy-paster error - should be corrected now @Jojain .

Jojain commented on doc/installation.rst:
```diff
@@ -7,9 +7,27 @@ CadQuery 2 is based on
+        :: Activate
+        cmd /K ""%USERPROFILE%/Miniforge3/Scripts/activate.bat" "%USERPROFILE%/Miniforge3""
+
+It might be worthwhile to consider using ``/NoScipts=0`` to add an activation shortcut.
```
typo ``/NoScipts=0`` -> ``/NoScripts=0``

It's however not clear how this relate to a shortcut and how to setup one ?

adam-urbanczyk replied:
Done, this line adds a shortcut to the start menu

Jojain commented:
I got the install successful. I've read very quickly what is miniforge, if I understand well (put aside what's under the hood) miniforge is miniconda with the default channel setup to `conda-forge` ? So one could install cadquery with the following :
`conda install -c cadquery cadquery=master` ?

Since I have also miniconda installed I don't know which one is activating but if I run 
`conda activate base` in a cmd prompt does it activate the miniforge env or I need to use your command line (for activation?)

jmwright commented:
@Jojain On Linux, I run something like `$HOME/miniforge/bin/activate` to make sure I have the right one (notice the path has `miniforge` and not `minconda`. That's assuming that miniconda is not already activated for the current shell.

jmwright commented:
I swapped Anaconda out for Miniforge on my main Ubuntu 20.04 install and it works fine. However, the solve still always fails on the Lubuntu 20.04 VM that I've also tried it on.  Below is what it always shows.

![Screenshot from 2022-02-08 08-13-43](https://user-images.githubusercontent.com/1015439/152994206-9a405ec1-acad-4d8c-a005-c98de37b38aa.png)

adam-urbanczyk commented:
> 

Can you share (link to)  the vm image?

jmwright commented:
VM image download is [here](https://cdimage.ubuntu.com/lubuntu/releases/20.04.3/release/lubuntu-20.04.3-desktop-amd64.iso). I ran it in VirtualBox.

lorenzncode commented:
@jmwright This could be an out of memory issue.  I've also experienced that with `conda` when allocating too little memory.  I've had it fail when allocating 4GB.  I couldn't reproduce that today at 4GB.  Here at 2GB (`dmesg`):

![Screenshot from 2022-02-08 23-11-49](https://user-images.githubusercontent.com/16394272/153120983-680a5d95-a62b-4905-8cfa-01ed6733e235.png)

One advantage of miniforge is it also provides mamba.  You might try replacing `conda install` with `mamba install` and see if it completes successfully.  

I checked RSS of the conda/mamba install process today on Ubuntu 20.04.
conda:   peak 2.7GiB  (separate install from above, with more memory allocated)
mamba: peak 280MiB

Jojain commented:
I don't really want to break my miniconda install to try miniforge alone, but I have been successful with the installation and activate on my Windows computer, so I think it's good (I've also read it's bundled directly with mamba ?)

jmwright commented:
If `mamba` uses less memory, should we consider using it by default instead of the `conda` command? I'm not familiar with `mamba`, so I don't know the implications.

Jojain commented:
I don't know about the memory but it's faster than conda and gives better error message when there is dependencies conflicts (however for some reason ``mamba activate myenv` doesn't work so I use both conda and mamba )

fedorkotov commented:
I have just nuked minicaonda and replaced it with mambaforge (condaforge variant with mamba preinstalled in base environment). It is fast. And `mamba activate envname` works. But importing cadquery fails see #990. Don't know if mamba or condaforge have anything to do with this problem though. My environment for cadquery in miniconda (that is now destroyed) worked but it was not updated for a long time.

adam-urbanczyk commented:
@jmwright @Jojain I think I fixed all the defects. Do you want to change from miniforge to mambaforge or leave as-is?

BTW: I cannot reproduce #990.

jmwright commented:
I just switched to mambaforge and it worked fine to create a CadQuery environment, and `import cadquery` worked as well. I'm on Linux though (Ubuntu 20.04).

jmwright commented:
I am also unable to reproduce #990 on Windows 11 64-bit with mambaforge. I'm ok with using mambaforge, but again, don't have a lot of experience with it. I have switch my systems over to it and will be using it as my primary way to handle my CadQuery environments moving forward.

adam-urbanczyk commented:
> I am also unable to reproduce #990 on Windows 11 64-bit with mambaforge. I'm ok with using mambaforge, but again, don't have a lot of experience with it. I have switch my systems over to it and will be using it as my primary way to handle my CadQuery environments moving forward.

OK, so I'd propose to stay with conda for now. The only known issue is the memory usage, I think that it is acceptable.

fedorkotov commented:
For me a workaround for #990 on Windows 8.1 was adding `-c conda-forge` to installation or environment creation command. No one seems to be able to reproduce it on more modern operating systems (Windows or Linux). Besides Windows 8.1 has only one more year of life left before end of extended support from M$. So this is not really an issue anymore.

adam-urbanczyk commented:
Alright merging with conda/miniforge, in the future we'll probably change it to mamba/mambaforge.
